### DISCUSSION TITLE
Shellに必要な機能

### DISCUSSION THREAD

**Discussion by hayashi-ay** [katataku/minishell#8] [General]
Shellに必要な機能の洗い出し。各機能の仕様の詳細や実装方針については別のDiscussionでするのが良さそう。

**Comment by hayashi-ay** [comment]
バックグラウンド実行、&をつけるやつは対応する必要ある？

**Reply by katataku** [reply]
課題に見つからないですね！
Anything that is not asked is not required.

**Reply by hayashi-ay** [reply]
そうしましょう。

**Reply by hayashi-ay** [reply]
かなりシンプルにいきましょう。その分、仕様まとめときたいですね。

こんな感じでdocsに

このシェルは以下の機能を備えている
- ヒアドキュメント
- etc

以下の機能には対応しない
- バックグラウンド実行
- etc

**Comment by hayashi-ay** [comment]
環境変数周りの処理はenvモジュール的な感じで切り出したいかも。pipexだとPATHから取得するのに特化していたけど。

**Reply by katataku** [reply]
例えばenv_getpathでPATHを返す、的なですかね。

他にenv関連でやることってあります？
export,unset用のgetter,setterをモジュールとして用意する感じですかね。

**Comment by katataku** [comment]
`1< Makefile cat`は対応必要！？

**Reply by katataku** [reply]
標準エラー出力の対応のこと忘れてた
